src/Compilers/CSharp/Portable/FlowAnalysis/PreciseAbstractFlowPass_Switch.cs

```diff
@@ -171,10 +171,19 @@ private void VisitPatternSwitchBlock(BoundPatternSwitchStatement node)
                 }
             }
 
-            // we always consider the default label reachable for flow analysis purposes.
+            // we always consider the default label reachable for flow analysis purposes
+            // unless there was a single case that would match every input.
             if (node.DefaultLabel != null)
             {
-                SetState(initialState.Clone());
+                if (node.SomeLabelAlwaysMatches)
```

Comment:
Will this only produce warnings in cases that already have errors or warnings?

Reply:
See the thread on the compat council email. The short answer is that there are circumstances that newly get errors:
``` c#
    switch (default(object))
    {
      case bool _:
      case true: // error: subsumed
      case false: // error: subsumed
        break;
    }
```
and circumstances that newly get warnings:
``` c#
    switch (1)
    {
      case 1 when true:
        break;
      default:
        break; // warning: unreachable
    }
```